### ISSUE TITLE
test_socks.py fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8665]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8665|
|Type|enhancement|
|Created|2016-07-24 12:57:30Z|
|Branch|https://github.com/twisted/twisted/tree/8665-rodrigc-socks|

If I run the following on Python 3, it fails:

```
python3 bin/trial twisted.test.test_socks
```

```
twisted.test.test_socks
  BindTests
    test_accessDenied ...                                               [ERROR]
    test_badSource ...                                                  [ERROR]
    test_eofLocal ...                                                   [ERROR]
    test_eofRemote ...                                                  [ERROR]
    test_simple ...                                                     [ERROR]
    test_socks4a ...                                                    [ERROR]
    test_socks4aFailedResolution ...                                    [ERROR]
  ConnectTests
    test_accessDenied ...                                               [ERROR]
    test_eofLocal ...                                                   [ERROR]
    test_eofRemote ...                                                  [ERROR]
    test_simple ...                                                     [ERROR]
    test_socks4aFailedResolution ...                                    [ERROR]
    test_socks4aSuccessfulResolution ...                                [ERROR]

===============================================================================
```

<details><summary>Searchable metadata</summary>

```
trac-id__8665 8665
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8665_rodrigc_socks 8665-rodrigc-socks
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469365050810460 1469365050810460
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/402

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset bb50a691b0c17bfc9c99e53e1d30def15cf969e4
```
#!CommitTicketReference repository="" revision="bb50a691b0c17bfc9c99e53e1d30def15cf969e4"
Merge pull request #402 from twisted/8665-rodrigc-socks

Author: rodrigc
Reviewer: alex
Fixes: #8665

Port twisted.protocols.socks to Python 3
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
